### PR TITLE
fix #8464 wrong root dir when -c is passed

### LINKED ISSUE #8464: -c flag produces the wrong root dir
### Scenario tested

```
tests/
       test.py
pytest.ini
pytest2.ini
```

The contents of `test.py`

```py
from pathlib import Path

def test_root(pytestconfig):
    assert Path(pytestconfig.rootpath).resolve().absolute() == Path('.').resolve().absolute()

```
`pytest.ini` and `pytest2.ini` are the same

+ When running  `pytest tests/` the test is **passing** (correct rootdir is printed):  `rootdir: D:\pytest, configfile: pytest.ini`
+ When running `pytest tests/ -c pytest2.ini` the test is **failing** (wrong rootdir and config are reported) :  `rootdir: D:\pytest\tests, configfile: ..\pytest2.ini`

Please let me know if I can submit a fix for this :)

### PR DESCRIPTION
**Pull request by elbehery95**
Fix for https://github.com/pytest-dev/pytest/issues/8464 including a doc update

@nicoddemus  let me know your thoughts and feedback

### CODE REVIEW

File: doc/en/reference/customize.rst [hunk 1]
```diff
@@ -160,10 +160,12 @@ Here is the algorithm which finds the rootdir from ``args``:
   ``setup.cfg`` in each of the specified ``args`` and upwards. If one is
   matched, it becomes the ``configfile`` and its directory becomes the ``rootdir``.
 
-- If no ``configfile`` was found, use the already determined common ancestor as root
+- If no ``configfile`` was found and no configuration argument is passed, use the already determined common ancestor as root
   directory. This allows the use of pytest in structures that are not part of
   a package and don't have any particular configuration file.
 
+- If no ``configfile`` was found and configuration argument is passed, use the configuration argument directory as root directory.
```
Comment (nicoddemus) [inline doc/en/reference/customize.rst:167]: Let's move this as a first item in this list, and reword it a bit:

```
- If ``-c`` is passed, use that as configuration file and its directory as ``rootdir``.
```

File: src/_pytest/config/findpaths.py [hunk 2]
```diff
@@ -176,7 +176,11 @@ def determine_setup(
         inipath: Optional[Path] = inipath_
         inicfg = load_config_dict_from_file(inipath_) or {}
         if rootdir_cmd_arg is None:
-            rootdir = get_common_ancestor(dirs)
```
Comment (nicoddemus) [inline src/_pytest/config/findpaths.py:179]: Actually if ``inifile`` is given, we should use its directory unconditionally as ``rootdir``, so this line should be:

```python
if rootdir_cmd_arg is None:
    rootdir = inipath.parent
```

File: testing/test_config.py [hunk 3]
```diff
@@ -1405,6 +1405,51 @@ def test_with_specific_inifile(
         assert inipath == p
         assert ini_config == {"x": "10"}
 
+    @pytest.mark.parametrize(
+        "tree,ini_file,args, expected_root",
```
Comment (nicoddemus) [inline testing/test_config.py:1409]: Let's use spaces here for readability:

```suggestion
        "tree, ini_file, args, expected_root",
```

File: testing/test_config.py [hunk 4]
```diff
@@ -1405,6 +1405,51 @@ def test_with_specific_inifile(
...
+            pytest.param(
+                ["tests/a/test1", "tests/b/test2"],
+                "tests/a/cfg.ini",
+                ["tests"],
+                "tests/a",
+                id="nested",
+            ),
+        ],
+    )
+    def test_with_args_and_specific_inifile(
```
Comment (nicoddemus) [inline testing/test_config.py:1434]: Thanks for being through here, but I think we only need a test similar to the one posted in the original issue:

```python
def test_explicit_config_file_sets_rootdir(tmp_path: Path, monkeypatch: pytest.MonkeyPatch) -> None:
    tests_dir = tmp_path / "tests"
    tests_dir.mkdir()

    monkeypatch.chdir(tmp_path)

    # No config file is explicitly given: rootdir is determined to be cwd.
    rootpath, found_inipath, *_ = determine_setup(None, [str(tests_dir)])
    assert rootpath == tmp_path
    assert found_inipath is None

    # Config file is explicitly given: rootdir is determined to be inifile's directory.
    inipath = (tmp_path / "pytest.ini")
    inipath.touch()
    rootpath, found_inipath, *_ = determine_setup(str(inipath), [str(tests_dir)])
    assert rootpath == tmp_path
    assert found_inipath == inipath
```

This should be enough to enforce the correct behavior.

**Review by nicoddemus — CHANGES_REQUESTED** [review]
Thanks a lot @elbehery95 for the PR!

However I think I wasn't very clear, but the semantics here should be a bit different: if an inifile is passed in the command-line, then we should always use its directory as rootdir. Please see my comments. 👍

File: src/_pytest/config/findpaths.py (hunk 2 above)
Comment (elbehery95) [reply src/_pytest/config/findpaths.py:179]: Thanks totally agree

File: testing/test_config.py (hunk 4 above)
Comment (elbehery95) [reply testing/test_config.py]: Much cleaner and on-point, replaced  `test_with_args_and_specific_inifile` with `test_explicit_config_file_sets_rootdir` thanks so much for clarification

File: changelog/8464.bugfix.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+``-c`` no longer yield wrong rootdir.
```
Comment (nicoddemus) [inline changelog/8464.bugfix.rst:1]:
```suggestion
``-c <config file>`` now also properly defines ``rootdir`` as the directory that contains ``<config file>``.
```

**Review by nicoddemus — APPROVED** [review]
Thanks a ton @elbehery95 for the contribution!

**Pull request merged**